src/basic/alloc-util.c

```diff
@@ -63,6 +64,22 @@ void* greedy_realloc(void **p, size_t *allocated, size_t need, size_t size) {
         if (!q)
                 return NULL;
 
+        bn = malloc_usable_size(q) / size;
```

Comment:
Maybe we should check for `size != 0` like I did in #12457

Or add an `assert(size > 0)`